### PR TITLE
Initial prow job configuration and OWNERS file adjustment

### PR DESCRIPTION
**Pull request by radoslawc**

### CODE REVIEW

File: .prow.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    run_if_changed: "(\\.go)$"
+    spec:
+      containers:
+      - image: golang:1.20.2
+        command:
+        - "/bin/sh"
+        - "-c"
+        - |
+          wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.52.2
+          bin/golangci-lint run -v
```
Comment (electrocucaracha) [inline .prow.yaml:43]: @radoslawc why don't you use the [golangci-lint official image](https://hub.docker.com/r/golangci/golangci-lint)

```suggestion
      - image: golangci/golangci-lint:v1.52
        command:
        - golangci-lint run -v
```

**Comment by johnbelamaric** [comment]
Prow is still not happy with the owners file, for some reason?

File: .prow.yaml (hunk 1 above)
Comment (johnbelamaric) [reply .prow.yaml:43]: @radoslawc will address in a follow-up PR

**Comment by johnbelamaric** [comment]
Manually merged to fix invalid OWNERS file.

**Pull request merged**